src/config/GeneralConfig.js

```diff
@@ -0,0 +1,193 @@
+'use strict';
+
+/**
+ * GeneralConfig
+ * @module
+ * 
+ * @description Configuration settings for various environments. 
+ * Adapted from general.json to avoid errors with manual version number updates.
+ * @author Chris Cole
+ * 
+ * */
+
+var VERSION = require('../_version.js');
+
+// Add an explicit value here to control how builds verify during the build automation testing.
+// Normally a build assumes one version prior. This can be used to specify a build number for
+// testing purposes in the CircleCI build. Leave null for default behavior
+var BUILD_VERSION_OVERRIDE = null;
+
+var VARIABLE_CURRENT_VERSION = "%CURRENT_VERSION%";
+var VARIABLE_PRIOR_VERSION = "%PRIOR_VERSION%";
+
+var configJson = {
+  "prod": {
+    "urls": {
+      "digitrustHostPath": "https://cdn.digitru.st/prod/%CURRENT_VERSION%/",
+      "digitrustRedirect": "https://cdn.digitru.st/prod/%CURRENT_VERSION%/redirect.html",
+      "digitrustIframe": "https://cdn.digitru.st/prod/%CURRENT_VERSION%/dt.html",
+      "digitrustIdService": "https://cdn-cf.digitru.st/id/v1",
+      "optoutInfo": "http://www.digitru.st/about-this-notice/",
+      "adblockCheck": "http://stats.aws.rubiconproject.com/"
+    },
+    "redirectInterval": {
+      "exp": 7
+    },
+    "cookie": {
+      "version": 2,
+      "producer": "1CrsdUNAo6",
+      "publisher": {
+        "domainKeyValue": "",
+        "pathKeyValue": "path=/;",
+        "maxAgeMiliseconds": 604800000,
+        "userObjectKey": "DigiTrust.v1.identity"
+      },
+      "digitrust": {
+        "domainKeyValue": "domain=.digitru.st;",
+        "pathKeyValue": "path=/;",
+        "maxAgeMiliseconds": 31536000000,
+        "userObjectKey": "DigiTrust.v1.identity",
+        "resetKey": "DeleteEverything",
+        "challenge": "DTChal",
+        "optout": "optout",
+        "optoutUser": {
+          "id": "",
+          "privacy": {
+            "optout": true
+          }
+        },
+        "errorUser": {
+          "error": true
+        }
+      }
+    },
+    "iframe": {
+      "timeoutDuration": 10000,
+      "postMessageOrigin": "https://cdn.digitru.st"
+    },
+    "crypto": {
+      "serverCryptoRate": 0.0
+    },
+    "logging": {
+      "enable": false,
+      "level": "ERROR"
+    },
+    "gvlVendorId": 64
+  },
+  "build": {
```

Comment:
what's the purpose of the `build` environment?

Reply:
Unit tests require a live environment to do the verification and testing. Since these test run prior to the release being deployed, we need to have a target that is already live in production. The test environment does not spin up to a stable URL to test against, so we run unit tests with iFrame pointed to "last known deployment" for the tests to run.

Reply:
gotcha